exonum_launcher/instances/default_spec_loader.py

```diff
@@ -0,0 +1,43 @@
+"""Default spec loader which will be attempted to be used in case
+if concrete loader was not provided for artifact."""
+
+from exonum_client.protobuf_loader import ProtobufLoader
+from exonum_client.module_manager import ModuleManager
+from exonum_client.proofs.encoder import build_encoder_function
+
+from exonum_launcher.configuration import Instance
+
+from .instance_spec_loader import InstanceSpecLoader, InstanceSpecLoadError
+
+
+class DefaultInstanceSpecLoader(InstanceSpecLoader):
+    """Default spec loader.
+    It attempts to load artifact proto files and recursively merge
+    Config message from provided instance config dict."""
+
+    def load_spec(self, loader: ProtobufLoader, instance: Instance) -> bytes:
+        try:
+            try:
+                # Try to load module (if it's already compiled) first.
+                service_module = ModuleManager.import_service_module(instance.artifact.name, "service")
+            except (ModuleNotFoundError, ImportError):
+                # If it's not compiled, load & compile protobuf.
+                loader.load_service_proto_files(instance.artifact.runtime_id, instance.artifact.name)
+                service_module = ModuleManager.import_service_module(instance.artifact.name, "service")
```

Comment:
It looks a little bit weird as for me. What if add some checker into the `Loader` for checking corresponding proto files ? And finally get something like this:

```python
if not loader.is_exist_service_proto_file(instance.artifact.name):
    loader.load_service_proto_files(instance.artifact.runtime_id, instance.artifact.name)

service_module = ModuleManager.import_service_module(instance.artifact.name, "service")
```

Reply:
Calling the code and test if it raises an exception is a common way tro check things in python, [for example](https://stackoverflow.com/a/319298).

However, it definetely makes sense to add method like `is_loaded` to `ProtobufLoader` and then rewrite this part.

I'll do it in the next PR to the light client.